src/Oscoin/Node/Options.hs

```diff
@@ -21,17 +21,18 @@ import           Network.Socket (HostName, PortNumber)
 import           Options.Applicative
 
 data Options = Options
-    { optHost           :: IP
-    , optGossipPort     :: PortNumber
-    , optApiPort        :: PortNumber
-    , optDiscovery      :: P2P.Disco.Options P2P.Disco.OptNetwork
-    , optBlockTimeLower :: Duration
-    , optPaths          :: Paths
-    , optEnvironment    :: Environment
-    , optMetricsHost    :: Maybe HostName
-    , optMetricsPort    :: Maybe PortNumber
-    , optEkgHost        :: Maybe HostName
-    , optEkgPort        :: Maybe PortNumber
+    { optHost                :: IP
+    , optGossipPort          :: PortNumber
+    , optApiPort             :: PortNumber
+    , optDiscovery           :: P2P.Disco.Options P2P.Disco.OptNetwork
+    , optBlockTimeLower      :: Duration
+    , optPaths               :: Paths
+    , optEnvironment         :: Environment
+    , optMetricsHost         :: Maybe HostName
+    , optMetricsPort         :: Maybe PortNumber
+    , optEkgHost             :: Maybe HostName
+    , optEkgPort             :: Maybe PortNumber
+    , optCreateKeysIfMissing :: Bool
```

Comment:
Jokes aside, perhaps an overkill thought: we already have in our `pathsParser` a `--keys` option to specify the desired `Filepath` for the keypair, and it feels slightly odd to have this "dangling" option for the creation of keys. I do wonder if we should create a nested record and treat "all-things-keys" as a single entity (but on the other hand, it will make the parser more complicated and the data structure more annoying to navigate, so perhaps it's not worth it).

Additionally, my reptilian brain is thinking that maybe we should somehow pick a different default path for the keys according to the `Environment` -- I can already see myself overriding production keys by accident when testing the `oscoin` client (to be precise on what I mean, having something like `~/.config/oscoin/mainnet/..` for production/mainnet and something different for test evironment(s))

Reply:
> "all-things-keys"

We treat all-things-paths as a single entity, which I think is more useful.

> different default path for the keys according to the Environment

You probably mean `Oscoin.P2P.Types.Network`. Good idea, actually.

Reply:
Unfortunately, this isn't possible easily: both the paths and the network are CLI options. If `Parser` was a monad, we _could_ make the paths parser depend on the network parser, such that it appends the network name to whatever you gave. This would be kind of odd though, because you explicitly give a `--keys` (f.ex.), and then it gets modified under the hood.

Since `Parser` _isn't_ a monad, we can't do this anyway. So how about naming the key _files_ after the network? Similarly for the blockstore -- one wouldn't specify the `*.db` file anymore, but only the directory.

Reply:
Would `ApplicativeDo` help as a last resort, or it's simply not worth it?

Reply:
No it wouldn't. You'd have to write:

```haskell
-- ...
optDiscovery <- discoParser
optPaths <- pathsParser paths (optNetwork optDiscovery)
-- ...
```

Which requires `Monad`